doc/reading/tma-modeling-in-depth.md

```diff
@@ -168,6 +176,298 @@ Another example is the following Figure 4:
+>
+> 1. $B < S$
+> 2. There exists $i_{t2} \in \mathbb{Z}$ that satisfies all of the following conditions:
+>    1. $0 < i_{t2}$
+>    2. $B \div e - 1 < i_{t2} < \mathrm{ceilDiv}(B, e)$
+>    3. $i_{t2} < S \div e$
+
+If $e$ divide $B$, then $\mathrm{ceilDiv}(B, e) = B / e = B \div e$,
+then $B \div e - 1 < i_{t2} < \mathrm{ceilDiv}(B, e)$ is not possible
+because there is no other integers between two consecutive integers.
+So (eq 9) can be simplified as
+
+> **(eq 10)** All the following conditions are satisfied:
+>
+> 1. $B < S$
+> 2. $e$ does not divide $B$
+> 3. There exists $i_{t2} \in \mathbb{Z}$ that satisfies all of the following conditions:
+>    1. $0 < i_{t2}$
+>    2. $B \div e - 1 < i_{t2} < \mathrm{ceilDiv}(B, e)$
+>    3. $i_{t2} < S \div e$
+
+which is equivalent to
+
+> **(eq 11)** All the following conditions are satisfied:
+>
+> 1. $B < S$
+> 2. $e$ does not divide $B$
+> 3. There exists $i_{t2} \in \mathbb{Z}$ that satisfies all of the following conditions:
+>    1. $1 \le i_{t2}$
+>    2. $\mathrm{ceilDiv}(B, e) - 1 \le i_{t2} < \mathrm{ceilDiv}(B, e)$
+>    3. $i_{t2} < S \div e$
+
+The condition (3.ii) in (eq 11) is achievable only if $i_{t2} = \mathrm{ceilDiv}(B, e) - 1$.
+So (eq 11) is equivalent to
+
+> **(eq 12)** All the following conditions are satisfied:
+>
+> 1. $B < S$
+> 2. $e$ does not divide $B$
+> 3. $\mathrm{ceilDiv}(B, e) \ge 2$
+> 4. $S > e \times (\mathrm{ceilDiv}(B, e) - 1)$
+
+which simplifies to
+
+> $e < B < S$ and $e$ does not divide $B$
+
+$\square$
+
+</details>
+
+Now, let's go back to the example in Figure 5.
+In this example, all the conditions of Theorem 6 are satisfied for the outer dimension,
+so strong correctness is unachievable. What if some condition is violated?
+
+If the element stride divided the box size, then the striding split would not create any hole at all.
+
+If the the tensor shape was `(4, 2)` instead of `(8, 2)`,
+which violated the condition that the box size is smaller than the tensor size,
+then the red items in Figure 6 would not exist, because items from `8` to `15` would now be out of boundary.
+
+If the element stride was `5` instead of `3`,
+the red items in Figure 6 would still exist without extra protextion,
+because the example of $i_3 = 0$, $i_4 = 4$, and $i_1 = 4$ mentioned above is still valid.
+However, by lowering smartly, we can add a check for the range of `I6` (i.e. $i_6 < 4$) to get rid of the red items.
+
+## The lowering strategy
+
+Theorem 6 is so powerful that it deserve a fancier name "**The fundamental theorem of TMA correctness**",
+or **FTTC** in short.
+It is powerful because the condition in it is both necessary and sufficient,
+and it directly tells us the correct lowering strategy:
+
+We should check the condition of FTTC.
+If the condition is true, then it is impossible to achieve strong correctness.
+So raise an error if strong correctness is required.
+If strong correctness is not required, generate code like below:
```

Comment:
If the strong correctness is not required, it means the weak correctness is sufficient, right? If so, why does it need additional predicates? Doesn't TMA automatically guarantee the weak correctness?

Reply:
It does automatically guarantee the weak correctness, but if there is a way to save some memory traffic, why not do it?

Reply:
I thought the only concern here was correctness.

Reply:
Modified the doc to be more clear on that